test/testpkg/src/subtract.cpp

```diff
@@ -0,0 +1,10 @@
+#include <R.h>
+#include <Rinternals.h>
+
+extern "C" SEXP subtract(SEXP a, SEXP b)
+{
+  SEXP result = PROTECT(allocVector(REALSXP, 1));
+  REAL(result)[0] = asReal(a) - asReal(b);
+  UNPROTECT(1);
+  return result;
+}
```

Comment:
Added this as a quick way to test basic C++ compilation during testing. Previously we only tested C and Fortran compilation. This would also be useful in r-builds testing, to catch issues like https://github.com/rstudio/r-builds/pull/39